content/en/docs/languages/net/instrumentation.md

```diff
@@ -63,320 +281,360 @@ using var tracerProvider = Sdk.CreateTracerProviderBuilder()
-      .AddAspNetCoreInstrumentation()
-      .AddConsoleExporter();
-  });
-```
+If a `TracerProvider` is not created, the OpenTelemetry APIs for tracing will
+use a no-op implementation and fail to generate data.
 
-This is also where you can configure instrumentation libraries.
+If you followed the instructions to [initialize the SDK](#initialize-the-sdk)
+above, you have a `TracerProvider` setup for you already. You can continue with
+[acquiring a tracer](#acquiring-a-tracer).
 
-Note that this sample uses the Console Exporter. If you are exporting to another
-endpoint, you'll have to use a different exporter.
 
 ### Setting up an ActivitySource
 
-Once tracing is initialized, you can configure an
+Anywhere in your application where you write manual tracing code should configure an
 [`ActivitySource`](/docs/concepts/signals/traces/#tracer), which will be how you
 trace operations with [`Activity`](/docs/concepts/signals/traces/#spans)
 elements.
 
-Typically, an `ActivitySource` is instantiated once per app/service that is
-being instrumented, so it's a good idea to instantiate it once in a shared
-location. It is also typically named the same as the Service Name.
+For example:
 
 ```csharp
-using System.Diagnostics;
 
 public static class Telemetry
 {
     //...
 
-    // Name it after the service name for your app.
-    // It can come from a config file, constants file, etc.
-    public static readonly ActivitySource MyActivitySource = new(TelemetryConstants.ServiceName);
+    public static readonly ActivitySource MyActivitySource = new("name", "version");
 
     //...
 }
 ```
+The values of name and version should uniquely identify the Instrumentation Scope, such as the package, module or class name. While the name is required, the version is still recommended despite being optional.
 
-You can instantiate several `ActivitySource`s if that suits your scenario,
-although it is generally sufficient to just have one defined per service.
+It’s generally recommended to define `ActivitySource` once per app/service that is been instrumented, but you can instantiate several `ActivitySource`s if that suits your scenario
 
-### Creating Activities
+In the case of the example app, the `ActivitySource` will be instantiated in the `program.cs` and then the same instance will be used in `DiceController.cs` and `Dice.cs`
 
-To create an [`Activity`](/docs/concepts/signals/traces/#spans), give it a name
-and create it from your
-[`ActivitySource`](/docs/concepts/signals/traces/#tracer).
+First we instantiate the activitySource in the `program.cs`:
 
-```csharp
-using var myActivity = MyActivitySource.StartActivity("SayHello");
+```csharp { hl_lines=["2-3"]}
 
-// do work that 'myActivity' will now track
-```
+// Register the ActivitySource as a singleton to have a single instance of activitySource across the application
+builder.Services.AddSingleton<ActivitySource>(new ActivitySource(serviceName, serviceVersion));
```

Comment:
this should not be done (`ActivitySource` is not a class defined/owned by this app, and should not be added to DI without wrapping inside a custom class on your own.)

See example from here : https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/examples/AspNetCore/Program.cs

Reply:
I ll update to include a new instrumentation class to wrap the ActivitySource